### PR TITLE
Http proxy

### PR DESCRIPTION
**Pull request by redragonx**

### CODE REVIEW

File: backend/src/net/proxy_controller.rs [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+
+                        let host = Self::host_addr(&req);
+
+                        match host {
+                            Ok(host_str) => {
+                                let dns_base =
+                                    host_str.split(':').next().unwrap_or_default().to_string();
+
+                                let docker_option = {
+                                    let mapping = docker_service_mapping.read_owned().await;
```
Comment (dr-bonez) [inline backend/src/net/proxy_controller.rs:63]:
```suggestion
                                    let mapping = docker_service_mapping.read().await;
```

File: backend/src/net/proxy_controller.rs [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+static RES_NOT_FOUND: &[u8] = b"503 Service Unavailable";
+static NO_HOST: &[u8] = b"No host header found";
+struct EmbassyHTTPServer {
+    docker_mapping: Arc<RwLock<BTreeMap<String, SocketAddr>>>,
+    shutdown: oneshot::Sender<()>,
+    handle: NonDetachingJoinHandle<()>,
+}
+impl EmbassyHTTPServer {
+    async fn new(listener_addr: SocketAddr) -> Result<Self, Error> {
```
[the remark this answers is not in the document]
Comment (redragonx) [reply backend/src/net/proxy_controller.rs:33]: Server::bind requires a SocketAddr with an IP and the port to bind to. So should `new` accept the IpAddr and a u16 for the port?

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: backend/src/backup/restore.rs [hunk 3]
```diff
@@ -232,11 +237,13 @@ pub async fn recover_full_embassy(
     .await?;
     secret_store.close().await;
 
-    Ok((
+    let embassy_recovered_data = (
         os_backup.tor_key.public().get_onion_address(),
         os_backup.root_ca_cert,
         async move {
             let rpc_ctx = RpcContext::init(ctx.config_path.clone(), disk_guid).await?;
+            NetController::setup_embassy_ui(rpc_ctx.clone()).await?;
```
[the remark this answers is not in the document]
Comment (redragonx) [reply backend/src/backup/restore.rs:245]: Do you have a way to activate that? So I can test.

File: backend/src/bin/avahi-alias.rs [hunk 4]
```diff
@@ -97,7 +97,6 @@ fn main() {
                 log_str_error("add CNAME record to Avahi entry group", res);
                 panic!("Failed to load Avahi services");
             }
-            eprintln!("Published {:?}", lan_address_ptr);
```
[the remark this answers is not in the document]
Comment (redragonx) [reply backend/src/bin/avahi-alias.rs:100]: ok

File: backend/src/error.rs [hunk 5]
```diff
@@ -22,7 +23,6 @@ pub enum ErrorKind {
     Utf8 = 13,
     ParseVersion = 14,
     IncorrectDisk = 15,
-    Nginx = 16,
```
[the remark this answers is not in the document]
Comment (redragonx) [reply backend/src/error.rs:25]: ok

**Review by dr-bonez — APPROVED** [review]

**Pull request merged**